sql/moz-fx-data-marketing-prod/adjust_derived/firefox_mobile_installs_v1/metadata.yaml

```diff
@@ -0,0 +1,17 @@
+description: Aggregated metrics from Adjust data
+friendly_name: Adjust Firefox mobile installs
+labels:
+  incremental: false
+  schedule: daily
+owners:
+  - rbaffourawuah@mozilla.com
+scheduling:
+  dag_name: bqetl_adjust
```

Comment:
One quick note on this, when the table is built, we will incrementally add new data to it and not do total table build each time. I did not know what the best practice is; i.e set the query up with a specific submission date first and then backfill after or create the initial table and incrementally add to it which is the idea I was shooting for with this etl. 

With that said, kindly recommend what is the best practice so I can refactor the query accordingly @sean-rose

Reply:
In order to safely make this incremental we'd need to know a couple things:
1. When are the `ga-mozilla-org-prod-001.Adjust.deliverable_*` tables created/updated?
   - This DAG is scheduled to run at 04:00 UTC, so if the table for the previous date isn't reliably created and complete by then, then we'd need to reschedule the DAG and/or implement an active check to detect when the data is available.
     - The check is probably advisable anyway, as whatever process is creating the `deliverable_*` tables might get delayed or break. An example of such a check is [ga_derived.www_site_empty_check_v1](https://github.com/mozilla/bigquery-etl/blob/main/sql/moz-fx-data-marketing-prod/ga_derived/www_site_empty_check_v1/query.sql).
2. Does each date-specific `deliverable_*` table reliably contain the complete data for that date, and only that date?
   - If there are sometimes straggler rows for other dates the ETL would need to account for that.

Assuming the answer to question 2 is yes, you'd add a condition like `_TABLE_SUFFIX = FORMAT_DATE('%Y%m%d', @submission_date)` (the `_TABLE_SUFFIX` gets the value of what the `*` in the table wildcard expression matched).

For the backfill case you could add an alternate condition to allow for the date parameter being null (e.g. `(_TABLE_SUFFIX = ... OR @submission_date IS NULL)`), or create an `init.sql` file (though then the query logic ends up being duplicated in `init.sql` and `query.sql` which is a maintenance hassle).  Personally I'd prefer the one-time hassle of the manual backfill rather than the ongoing query logic maintenance hassle.

Reply:
After getting access to the `ga-mozilla-org-prod-001.Adjust.deliverable_*` tables I can answer my own questions from above:
1. When are the `ga-mozilla-org-prod-001.Adjust.deliverable_*` tables created/updated?
   - They appear to get created by around 12:00 UTC the following day.
   - However, they also appear to continue to get updated for nine days after that (i.e. 10 days after the `date`).
     - If you're OK with the ETL table only containing data < 10 days ago we could use the `date_partition_offset` config to account for this.
     - Or if having recent data is important, there is a way to handle this delayed data in an incremental ETL including recent data, but it's not straightforward and IMO that'd be more complication than it's worth.
2. Does each date-specific `deliverable_*` table reliably contain the complete data for that date, and only that date?
   - Yes.

Also, I noticed there isn't a table for 2023-04-04, so apparently dates can be missing.

So given all that and the relatively small size of the data being processed (>=2022 is just 7.7 GB) I'd recommend keeping this rebuilding the table each day, but with an overall filter on `_TABLE_SUFFIX` so it scans less data (I'll add that as a separate comment/suggestion).

Reply:
Based on this information, I think a full table rebuild each run will be the right approach here as I do not really know how the underlying table gets updated. I will make the necessary changes and check them in accordingly